rholang/src/main/scala/BlockRandomSeed.scala

```diff
@@ -0,0 +1,34 @@
+import coop.rchain.crypto.PublicKey
+import coop.rchain.crypto.hash.Blake2b256.hash
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import scodec.bits.ByteVector
+import scodec.{Codec, TransformSyntax}
+import scodec.codecs.{bytes, uint16, uint8, ulong, utf8, variableSizeBytes}
+
+final case class BlockRandomSeed(
+    shardId: String,
+    blockNumber: Long,
+    sender: PublicKey,
+    validatorPublicKey: PublicKey,
+    preStateHash: Blake2b256Hash
+)
+
+object BlockRandomSeed {
+  private val codecPublicKey: Codec[PublicKey] = variableSizeBytes(uint8, bytes)
+    .xmap[PublicKey](bv => PublicKey(bv.toArray), pk => ByteVector(pk.bytes))
+  private val codecBlakeHash: Codec[Blake2b256Hash] =
+    variableSizeBytes(uint16, bytes)
+      .xmap[Blake2b256Hash](bv => Blake2b256Hash.fromByteVector(bv), _.bytes)
```

Comment:
I found this codec already defined

https://github.com/rchain/rchain/blob/44b6febd85277df1a1e34c83752b84d342f62327/rspace/src/main/scala/coop/rchain/rspace/hashing/Blake2b256Hash.scala#L82-L102

Maybe we can reuse it?

Reply:
Yes, I think it's better to use this codec. Thanks)